File: aScharfe/ArabicRomanKata_Python/ArabicRomanKata_Python/UnitTest/ArabicRomanKata_Test.py
import unittest
from Helper.CheckInputValue import CheckInputValue
from Converter.ArabicRomanConverter import ArabicRomanConverter


class ArabicRomanKataTest(unittest.TestCase):
    def test_CheckInputValue(self):
        self.assertTrue(CheckInputValue.CheckValueIsValid("1"),)
        self.assertTrue(CheckInputValue.CheckValueIsValid("12"))
        self.assertTrue(CheckInputValue.CheckValueIsValid("123"))
        self.assertTrue(CheckInputValue.CheckValueIsValid("1234"))
        self.assertTrue(CheckInputValue.CheckValueIsValid("I"))
        self.assertTrue(CheckInputValue.CheckValueIsValid("XII"))
        self.assertTrue(CheckInputValue.CheckValueIsValid("CXXIII"))
        self.assertTrue(CheckInputValue.CheckValueIsValid("MCCXXXIV"))
        self.assertTrue(CheckInputValue.CheckValueIsValid("MMMCMXCIX"))
        
        self.assertFalse(CheckInputValue.CheckValueIsValid(""))
        self.assertFalse(CheckInputValue.CheckValueIsValid("0"))
        self.assertFalse(CheckInputValue.CheckValueIsValid("4000"))
        self.assertFalse(CheckInputValue.CheckValueIsValid("12345"))
        self.assertFalse(CheckInputValue.CheckValueIsValid("123d"))
        self.assertFalse(CheckInputValue.CheckValueIsValid("d123"))

        self.assertFalse(CheckInputValue.CheckValueIsValid("MMMMMM"))
        self.assertFalse(CheckInputValue.CheckValueIsValid("P"))

    def test_ArabicRomanConverter(self):
        self.assertEqual(ArabicRomanConverter.Convert(""),"   Cannot be converted.","Convert empty string")
        self.assertEqual(ArabicRomanConverter.Convert("0"),"","Convert 0")

        self.assertEqual(ArabicRomanConverter.Convert("1"),"I","Convert 1")
        self.assertEqual(ArabicRomanConverter.Convert("1974"),"MCMLXXIV","Convert 1974")
        self.assertEqual(ArabicRomanConverter.Convert("3999"),"MMMCMXCIX", "Convert 3999")
        
        
        self.assertEqual(ArabicRomanConverter.Convert("I"),1,"Convert I")
        self.assertEqual(ArabicRomanConverter.Convert("MMMCMXCIX"),3999,"Convert MMMCMXCIX")
        self.assertEqual(ArabicRomanConverter.Convert("MCMLXXIV"),1974,"Convert MCMLXXIV")
        self.assertEqual(ArabicRomanConverter.Convert("mcmlxxiv"),1974,"Convert MCMLXXIV")



if __name__ == '__main__':
    unittest.main()
        

File: aScharfe/ArabicRomanKata/ArabicRomanKata_UnitTest/ArabicRomanKataTest.cs
using System;
using ArabicRomanKata.Converter;
using ArabicRomanKata.Enum;
using ArabicRomanKata.Helper;
using Microsoft.VisualBasic;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace ArabicRomanKataUnitTest
{
    [TestClass]
    public class ArabicRomanKataTest
    {

        [TestMethod]
        public void TestCheckInputValue()
        {
            Assert.IsTrue("1".CheckValueIsValid());
            Assert.IsTrue("12".CheckValueIsValid());
            Assert.IsTrue("123".CheckValueIsValid());
            Assert.IsTrue("1234".CheckValueIsValid());
            Assert.IsTrue("I".CheckValueIsValid());
            Assert.IsTrue("XII".CheckValueIsValid());
            Assert.IsTrue("CXXIII".CheckValueIsValid());
            Assert.IsTrue("MCCXXXIV".CheckValueIsValid());
            Assert.IsTrue("MMMCMXCIX".CheckValueIsValid());

            Assert.IsFalse("".CheckValueIsValid());
            Assert.IsFalse("0".CheckValueIsValid());
            Assert.IsFalse("4000".CheckValueIsValid());
            Assert.IsFalse("12345".CheckValueIsValid());
            Assert.IsFalse("123d".CheckValueIsValid());
            Assert.IsFalse("d123".CheckValueIsValid());

            Assert.IsFalse("MMMMMM".CheckValueIsValid());
            Assert.IsFalse("P".CheckValueIsValid());
        }

        [TestMethod]
        public void TestArabicToRoman()
        {
            Assert.AreEqual("",0.ConvertArabicToRoman());
            Assert.AreEqual("I",1.ConvertArabicToRoman());
            Assert.AreEqual("MCMLXXIV", 1974.ConvertArabicToRoman());
            Assert.AreEqual("MMMCMXCIX", 3999.ConvertArabicToRoman());
        }

        [TestMethod]
        public void TestRomanToArabic()
        {
            Assert.AreEqual(0,"".ConvertRomanToArabic());
            Assert.AreEqual(1,"I".ConvertRomanToArabic());
            Assert.AreEqual(1974, "MCMLXXIV".ConvertRomanToArabic());
            Assert.AreEqual(3999, "MMMCMXCIX".ConvertRomanToArabic());
            Assert.AreEqual(1974, "mcmlxxiv".ConvertRomanToArabic());
        }

        [TestMethod]
        public void TestConversionDirection()
        {
            Assert.AreEqual(ConversionDirection.Unknown, "".GetConversionDirection());
            Assert.AreEqual(ConversionDirection.Unknown, "P".GetConversionDirection());
            Assert.AreEqual(ConversionDirection.ToArabic, "I".GetConversionDirection());
            Assert.AreEqual(ConversionDirection.ToRoman, "1".GetConversionDirection());
        }
    }
}
